src/queries/lendr-factory-contracts.ts

```
@@ -0,0 +1,167 @@
+    functionName: 's_totalRentals',
+  });
+
+  const { data: feeBps, refetch: refetchFeeBps } = useReadContract({
+    address: LENDR_FACTORY_CONTRACT_ADDRESS,
+    abi: LENDR_FACTORY_ABI,
+    functionName: 's_feeBps',
+  });
+
+  // Function to get collateral rental agreement by ID
+  const getCollateralRentalAgreement = (rentalId: number) => {
+    const { data: agreement, refetch } = useReadContract({
```

Biome (2.1.2)

[error] This hook is being called from a nested function, but all hooks must be called unconditionally from the top-level component.

For React to preserve state between calls, hooks needs to be called unconditionally and always in the same order.
See https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

(lint/correctness/useHookAtTopLevel)